### ISSUE TITLE
Phone number cannot be edited when the state/province(Region id) field contains value in admin order address Magento 2.4.5 

### ISSUE THREAD

**Issue by Dev11Developer** [magento/magento2#37608]
### Preconditions and environment

- Magento version 2.4.5 installed
- PHP version 7.4.x or 8.1.x 

### Steps to reproduce

1.  Go to the order view on the admin side
2.  Edit billing/shipping address
3.  Edit the phone number separately when the State/Province field has a value

### Expected result

The phone number is updated successfully for the order 

### Actual result

When we update the phone number individually it does not save the data in the database but when you update the phone number with other address values like name, postcode or city etc, the phone number is successfully updated to the order address.
You can only update the phone number individually when the state/province(Region id) doesn't contain any value.

### Additional information

Video for the issue
https://www.awesomescreenshot.com/video/18275512?key=c518e7792a503959336b817bcb10d44f

### Triage and priority

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [ ] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

**Comment by engcom-Bravo** [comment]
@magento give me 2.4-develop instance

**Comment by engcom-Bravo** [comment]
Hi @Dev11Developer,

Thank you for reporting and collaboration.

Verified the issue on Magento 2.4-develop instance and the issue is not reproducible.Kindly refer the attached video.


**Steps to reproduce**

- Go to the order view on the admin side
- Edit billing/shipping address
- Edit the phone number separately when the State/Province field has a value

We are able to edit the phone number successfully after placing an order.

https://github.com/magento/magento2/assets/51680745/d69f9b8c-7ca4-4d7f-a5ad-e76c541c2aa5

Kindly recheck the behaviour on Magento 2.4-develop instance and elaborate steps to reproduce if the issue is still reproducible.


Thanks.

**Comment by Dev11Developer** [comment]
No, please just change the phone number and save it. If you edit and save it with other fields it will update the phone number but when you update the phone number alone it doesn't update.
https://www.awesomescreenshot.com/video/18312998?key=08263d77fa1dfe3ad33710e93372fd9b

**Comment by engcom-Bravo** [comment]
@magento give me 2.4-develop instance

**Comment by engcom-Bravo** [comment]
Hi @Dev11Developer,

Thanks for your update.

Verified the issue on Magento 2.4-develop instance and the issue is reproducible.Kindly refer the attached video.


**Steps to reproduce**

- Go to the order view on the admin side
- Edit billing/shipping address
- Edit the phone number separately when the State/Province field has a value.


The issue is only with digit `0` we are not able to update the phone number with `0` for the first digit of phone number but we are able to update the phone number with `0` other than first digit.



https://github.com/magento/magento2/assets/51680745/1f3b33e8-4a12-4fe9-bab8-18fba08c6184




We are able to update the phone number for other digits successfully.


Hence Confirming this issue.


Thanks.

**Comment by engcom-Bravo** [comment]
Hi @Dev11Developer,

Thanks for your reporting and collaboration.

We have verified the issue in Latest 2.4-develop instance and the issue is no more reproducible.Kindly refer the attached video.


https://github.com/user-attachments/assets/b0fcd300-ca23-4060-9b1e-a77af5a767a3



The phone number is updated successfully for the order


Hence we are closing this issue.

Thanks.
